ingest/annotations.py

```diff
@@ -9,27 +9,38 @@
 
 import abc
 from collections import defaultdict
+import logging
 
 import pandas as pd
 from bson.objectid import ObjectId
 
 try:
     # Used when importing internally and in tests
     from ingest_files import IngestFiles
+    from monitor import setup_logger
+
 except ImportError:
     # Used when importing as external package, e.g. imports in single_cell_portal code
     from .ingest_files import IngestFiles
+    from .monitor import setup_logger
 
 
 class Annotations(IngestFiles):
     __metaclass__ = abc.ABCMeta
 
+    errors_logger = setup_logger(
+        __name__ + '_errors', 'errors.txt', level=logging.ERROR
+    )
+
     def __init__(
         self, file_path, allowed_file_types, study_id=None, study_file_id=None
     ):
         IngestFiles.__init__(self, file_path, allowed_file_types)
         if study_id is not None:
             self.study_id = ObjectId(study_id)
+            self.extra_log_params = {'study_id': self.study_id, 'duration': None}
```

Comment:
Worth noting, but not urgent: these event properties are subtly incompatible with Mixpanel analytics.  Mixpanel uses:

* camelCase instead of snake_case
* `perfTime` and not `duration` (nor `latency`) to track time intervals like this
* Study accession instead of study ID

These logs are not yet sent to Mixpanel, but if/when they are, they would ideally be compatible with SCP logging from other sources (client-side JS, server-side Ruby).

Reply:
created SCP-2591 to track this concern.